### PR TITLE
Add options to send notification to nats using jwt / nkey or using creds

### PR DESCRIPTION
**Pull request by HugoDab**
## Description

This Pull Request enables minio to connect to a nats server using jwt and nkey or credentials file.

## Motivation and Context

JWT, nkey and creds are useful to configure permissions on nats. Unfortunately, before this Pull Request, minio was not able to connect to nats this way.

## How to test this PR?

You have to create a notification for nats, for example by using :
`mc admin config set myminio notify_nats`
With the arguments, `jwt=` and `nkey_seed=` or `creds=`

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [X] Documentation updated
- [ ] Unit tests added/updated

### CODE REVIEW

**Review by harshavardhana — CHANGES_REQUESTED** [review]
why does NATS have 100 different ways to authenticate?

**Comment by HugoDab** [comment]
Nkey/jwt or creds are the efficient way to enable RBAC in NATS.

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by harshavardhana** [comment]
> Nkey/jwt or creds are the efficient way to enable RBAC in NATS.

Why are they efficient and why is that not the default choice? - every 6months some new technique gets introduced.

First, it was username/password
Next, it was token-based
Next, it was client TLS (mTLS)

and now this? 

How is it that it's efficient and none of the other options are not? what is the benchmark for such efficiency? 

We are the maintainers here it looks like a cockpit of authentication options with no clear idea what is correct and what is not. I would rather not have it in our project if it's used by some users out there who know how to configure it. 

Enlighten me why the other 3 existing options are not great?

**Comment by HugoDab** [comment]
The user/password and token-based authentication are only available in a static way in the configuration file of Nats, whereas the 2.0 version of NATS server introduced the idea of decentralized authentication based on JSON Web Tokens (JWT). Clients interact with this new scheme using a user JWT and corresponding NKey private key.

In a multitenancy context, it enables to create a notification per bucket to a dedicated subject for user jwt/nkey. It allows a fine grained RBAC control with a dynamic user jwt/nkey creation:
https://docs.nats.io/developing-with-nats/security/creds

The usage of JWT enables more efficient to control user authorizations:
https://docs.nats.io/nats-tools/nsc/basics#user-authorization

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by harshavardhana** [comment]
Still relevant, PR is backlogged as we get other PRs merged.

**Comment by cameronbraid** [comment]
This would be great to add.

Please also support JWT without nkey_seed as the JWT can be configured as a bearer

File: internal/event/target/nats.go [hunk 1]
```diff
@@ -150,6 +160,18 @@ func (n NATSArgs) connectNats() (*nats.Conn, error) {
 	if n.Token != "" {
 		connOpts = append(connOpts, nats.Token(n.Token))
 	}
+	if n.JWT != "" && n.NkeySeed != nil {
```
Comment (cameronbraid) [inline internal/event/target/nats.go:163]: Please support JWT without NkeySeed, as the JWT can be configured as a bearer

```
userSignature := func(nonce []byte) ([]byte, error) {
	if n.NkeySeed = nil {
		return nil, nil
	}

	return sigHandler(nonce, n.NkeySeed)
}
```

**Comment by harshavardhana** [comment]
Please rebase with the latest master and resubmit if this still needs to be merged. Thank you.

**Pull request closed without merging**